### ISSUE TITLE
Terrafirmacraft

### ISSUE THREAD

**Issue by SpoilersMC** [makamys/Satchels#4]
If youre still working on this mod- i'd like to request terrafirmacraft(and plus) support. 
currently incompatible because they replace the vanilla inventory and Satchels.onGameTypeChanged and a few other places assume you can always cast player.inventoryContainer to ContainerSatchels

if youre no longer working on this mod would you have any problem with me making a pr for this?

**Comment by makamys** [comment]
This has been requested before, but I haven't been invested enough in TFC to add it. Especially since ceramic jugs already let you increase your storage capacity in the early game, which is where Satchels is (in my opinion) the most useful.

I'd accept a PR, though. But do note that the GUI would have to be redesigned, since TFC's is so different. In particular, the 8 pouch slots would have to be rearranged into two columns of 4 instead of a single column of 8 to fit under the things TFC adds on the side. It would look something like this fully upgraded (in TFC+):
![](https://user-images.githubusercontent.com/13667520/148659583-de53692f-b51f-486f-876b-c7e15c60707d.png)

There should also be an option to only let pouches hold small (in TFC terms) items.

**Comment by SpoilersMC** [comment]
yeah i was also thinking of adding a config to reduce the normal inventory size by a row and using satchels as a way to 'unlock' it similar to unlocking the 3x3 inventory crafting table- but that'd be a bit of work so we'll see if i get around to it
